classes/TaskRunner/Upgrade/UninstallIncompatibleModules.php

```diff
@@ -0,0 +1,158 @@
+        'blockcart',
+        'blockcategories',
+        'blockcms',
+        'blockcmsinfo',
+        'blockcontact',
+        'blockcurrencies',
+        'blocklanguages',
+        'blocklayered',
+        'blockmyaccount',
+        'blocknewsletter',
+        'blocksearch',
+        'blocksocial',
+        'blocktopmenu',
+        'blockuserinfo',
+        'cheque',
+        'homefeatured',
+        'homeslider',
+        'onboarding',
+        'socialsharing',
+        'vatnumber',
+        'blockadvertising',
+        'blockbestsellers',
+        'blockcustomerprivacy',
+        'blocklink',
+        'blockmanufacturer',
+        'blocknewproducts',
+        'blockpermanentlinks',
+        'blockrss',
+        'blocksharefb',
+        'blockspecials',
+        'blocksupplier',
+        'blockviewed',
+        'crossselling',
+        'followup',
+        'productscategory',
+        'producttooltip',
+        'mailalert',
+        'blockcontactinfos',
+        'blockfacebook',
+        'blockmyaccountfooter',
+        'blockpaymentlogo',
+        'blockstore',
+        'blocktags',
+        'blockwishlist',
+        'productpaymentlogos',
+        'sendtoafriend',
+        'themeconfigurator',
+    ];
+
+    public function run()
+    {
+        $this->next = 'upgradeDb';
+        if (version_compare($this->container->getState()->getInstallVersion(), '1.7.0.0', '<')) {
+            $this->logger->info(
+                $this->translator->trans(
+                    'No incompatible module to uninstall. Now upgrading database...',
+                    array(),
+                    'Modules.Autoupgrade.Admin'
+                )
+            );
+
+            return true;
+        }
+        $modulePath = $this->container->getProperty(UpgradeContainer::PS_ROOT_PATH)
+            . DIRECTORY_SEPARATOR . 'modules' . DIRECTORY_SEPARATOR;
+
+        if (!is_dir($modulePath)) {
+            throw (new UpgradeException(
+                $this->translator->trans(
+                    '[ERROR] %dir% does not exist or is not a directory.',
+                    array('%dir%' => $modulePath),
+                    'Modules.Autoupgrade.Admin'
+                )
+            ))
+                ->addQuickInfo(
+                    $this->translator->trans(
+                        '[ERROR] %s does not exist or is not a directory.',
+                        array($modulePath),
+                        'Modules.Autoupgrade.Admin'
+                    )
+                )
+                ->setSeverity(UpgradeException::SEVERITY_ERROR);
+        }
+
+        foreach (scandir($modulePath) as $moduleName) {
+            if (
+                in_array($moduleName, static::$incompatibleModules)
+                && file_exists($modulePath . $moduleName . DIRECTORY_SEPARATOR . $moduleName . '.php')
+            ) {
+                $this->logger->info('Uninstalling module ' . $moduleName);
+                \Module::getInstanceByName($moduleName)->uninstall();
```

Comment:
I'm wondering: maybe `\Module::getInstanceByName($moduleName)` might return something wrong if module is faulty? Maybe check this is a module instance and not `false` or `null` before calling `uninstall()`

Also `uninstall` does it return a result? Maybe it can fail.